packages/Takeover/src/Takeover.js

```diff
@@ -0,0 +1,116 @@
+import EscListener from "./components/EscListener";
+import Wrapper from "./components/Wrapper";
+import Header from "./components/Header";
+import LockBodyScroll from "./components/LockBodyScroll";
+import { animationDuration } from "./tokens";
+import extractChildren from "./helpers/extractChildren";
+
+const propTypes = {
+  /** The content for the Takeover. */
+  children: PropTypes.node.isRequired,
+
+  /** Control the visibility of the takeover */
+  isOpen: PropTypes.bool.isRequired,
+
+  /** Callback triggered when the takeover needs to be close */
+  onClose: PropTypes.func,
+
+  /** Callback once the takeover has been opened event */
+  onAfterOpen: PropTypes.func,
+
+  /** Callback once the takeover has been closed event */
+  onAfterClose: PropTypes.func,
+
+  /** Render the takeover inline */
+  isInline: PropTypes.bool,
+
+  /** Takeover will capture and will not propagate event after pressing ESC. Helps in case we have many overlapped components. */
+  shouldStopEscapePropagation: PropTypes.bool,
+};
+
+const defaultProps = {
+  isInline: false,
+  shouldStopEscapePropagation: false,
+  onAfterClose: () => {},
+  onClose: () => {},
+  onAfterOpen: () => {},
+};
+
+const Portal = ({ children, active }) =>
+  active ? ReactDOM.createPortal(children, document.body) : <React.Fragment>{children}</React.Fragment>;
+
+const StyledWrapper = styled(Wrapper)`
+  display: flex;
+  flex-direction: column;
+`;
+
+const StyledHeader = styled(Header)`
+  flex: none;
+`;
+
+const ContentWrapper = styled.div`
+  overflow-y: auto;
+  flex: 1 1 auto;
+`;
+
+const Takeover = ({ isOpen, onClose, isInline, shouldStopEscapePropagation, onAfterClose, onAfterOpen, ...props }) => {
+  const refWrapper = React.useRef(null);
+
+  function handleTransitionEnter(node) {
+    // https://github.com/reactjs/react-transition-group/blob/6dbadb594c7c2a2f15bc47afc6b4374cfd73c7c0/src/CSSTransition.js#L44
+    node.scrollTop;
+  }
+
+  const {
+    "Takeover.FocusTrap": focusTrapExtracted,
+    "Takeover.Header": headerExtracted,
+    "Takeover.Content": contentExtracted,
+    children,
+  } = extractChildren(props.children, ["Takeover.FocusTrap", "Takeover.Header", "Takeover.Content"]);
+
+  const extendedFocusTrapOptions = focusTrapExtracted ? focusTrapExtracted.props : {};
+
+  const focusTrapOptions = {
+    fallbackFocus: () => document.createElement("div"),
+    ...extendedFocusTrapOptions,
+  };
+
+  return (
+    <>
+      {isOpen && <EscListener on={onClose} shouldStopEscapePropagation={shouldStopEscapePropagation} />}
+      {isOpen && <LockBodyScroll />}
+      <Portal active={!isInline}>
+        <Transition
+          mountOnEnter
+          unmountOnExit
+          in={isOpen}
+          timeout={animationDuration}
+          onEnter={handleTransitionEnter}
+          onEntered={onAfterOpen}
+          onExited={onAfterClose}
+        >
+          {state => (
+            <FocusTrapLibrary active={!isInline} focusTrapOptions={focusTrapOptions}>
```

Comment:
Even an `inline` Takeover should trap the focus, shouldn't it?

Reply:
I just copied this behavior from SidePanel. Should SidePanel trap the focus in inline mode too?

Reply:
mmm, I think when the sidepanel is inline is a little more complex because we have an array of sidepanels, so maybe we will need to add extra code to trap the focus across all the sidepanels, so far that is not implemented and didn't think about that until now. 

Maybe the takeover shouldn't care if it's inline or not? 
cc @mikrotron

Reply:
Hm. I'm not sure if we need to have an inline option or not for Takeover... _maybe_ there's a reason to keep it's code inline in the DOM, rather than a portal appended to the end...but I can't really think of why you would _need_ that.  

For the SidePanel though , you may want an "inline" implementation that does _NOT_ trap focus, since the user can still see and interact with other page content.  The Takeover on the other hand blocks everything else out (it is "modal").

Reply:
@mikrotron @nahumzs I don't see any reason to have inline mode for Takeover. WDYT if I completely remove `inline` prop?

Reply:
Yes, let's remove it @DanilAgafonov